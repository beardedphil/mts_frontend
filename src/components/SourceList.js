import React from 'react';
import Source from './Source';

const SourceList = ({ sources = [] }) => {
    return (
        <div className="source-list">
            {
                sources.length === 0 ? (
                    <p>No sources found.</p>
                ) : (
                    sources.map((source) => {
                        return (
                            <Source key={source.id} source={source} />
                        );
                    })
                )
            }
        </div>
    );
};

export default SourceList;
