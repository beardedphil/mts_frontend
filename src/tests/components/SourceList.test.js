import React from 'react';
import { shallow } from 'enzyme';
import SourceList from '../../components/SourceList/SourceList';
import sources from '../fixtures/sources';

test('should render SourceList correctly with no sources', () => {
    const wrapper = shallow(<SourceList sources={[]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render SourceList correctly with sources', () => {
    const wrapper = shallow(<SourceList sources={ sources } />);
    expect(wrapper).toMatchSnapshot();
});
