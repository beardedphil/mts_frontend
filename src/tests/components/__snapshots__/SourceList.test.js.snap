// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SourceList correctly with no sources 1`] = `
<div
  className="source-list"
>
  <p>
    No sources found.
  </p>
</div>
`;

exports[`should render SourceList correctly with sources 1`] = `
<div
  className="source-list"
>
  <Source
    key="1"
    source={
      Object {
        "id": 1,
        "proper_source": "CNN",
        "source": "cnn",
      }
    }
  />
  <Source
    key="2"
    source={
      Object {
        "id": 2,
        "proper_source": "Slate",
        "source": "slate",
      }
    }
  />
  <Source
    key="3"
    source={
      Object {
        "id": 3,
        "proper_source": "MSNBC",
        "source": "msnbc",
      }
    }
  />
  <Source
    key="4"
    source={
      Object {
        "id": 4,
        "proper_source": "Fox News",
        "source": "fox news",
      }
    }
  />
</div>
`;
